conf/autopilot/rotorcraft_autopilot.xml

```diff
@@ -2,93 +2,139 @@
+      <dl_setting var="autopilot_mode_auto2" min="2" step="1" max="3" module="autopilot" values="NAV|GUIDED"/>
+    </settings>
 
     <control_block name="set_actuators">
       <call fun="SetRotorcraftCommands(stabilization_cmd, autopilot_in_flight, autopilot_motors_on)"/>
       <call fun="SetActuatorsFromCommands(commands, autopilot_mode)"/>
     </control_block>
 
     <control_block name="attitude_loop">
-      <call fun="SetAttitudeFromRC(rc_values)"/>
-      <call fun="AddAttitudeFromFMS()" cond="fms.enabled"/>
+      <call fun="stabilization_attitude_read_rc(autopilot_in_flight, FALSE, FALSE)"/>
       <call fun="stabilization_attitude_run(autopilot_in_flight)"/>
     </control_block>
 
+    <control_block name="throttle_direct">
+      <call fun="guidance_v_read_rc()"/>
+      <call fun="guidance_v_z_sp = stateGetPositionNed_i()->z"/>
+      <call fun="stabilization_cmd[COMMAND_THRUST] = guidance_v_rc_delta_t"/>
+    </control_block>
+
     <control_block name="altitude_loop">
-      <call fun="SetAltitudeFromFMS()" cond="fms.enabled"/>
       <call fun="gv_update_ref_from_z_sp(guidance_v_z_sp)"/>
       <call fun="run_hover_loop(autopilot_in_flight)"/>
-      <call fun="SaturateThrottle(rc_values)"/>
+      <!--call fun="SaturateThrottle(rc_values)"/-->
     </control_block>
 
     <exceptions>
       <exception cond="too_far_from_home" deroute="HOME"/>
+      <exception cond="kill_switch_is_on()" deroute="KILL"/>
     </exceptions>
 
-    <mode name="ATTITUDE" start="stabilization_attitude_enter()">
-      <select cond="$DEFAULT_MODE"/>
+    <mode name="ATTITUDE_DIRECT" start="stabilization_attitude_enter()" shortname="ATT">
       <select cond="RCMode0()"/>
-      <control freq="512">
+      <control>
         <call_block name="attitude_loop"/>
-        <call fun="SetThrottleFromRC(rc_values)"/>
+        <call_block name="throttle_direct"/>
         <call_block name="set_actuators"/>
       </control>
       <exception cond="RCLost()" deroute="FAILSAFE"/>
     </mode>
 
-    <mode name="VERTICAL" start="stabilization_attitude_enter()|guidance_v_enter()">
+    <mode name="ATTITUDE_Z_HOLD" start="stabilization_attitude_enter()|guidance_v_z_enter()" shortname="A_ZH">
       <select cond="RCMode1()"/>
-      <control freq="512">
+      <control>
         <call_block name="attitude_loop"/>
         <call_block name="altitude_loop"/>
         <call_block name="set_actuators"/>
       </control>
       <exception cond="RCLost()" deroute="FAILSAFE"/>
     </mode>
 
-    <mode name="NAV" start="guidance_h_nav_enter()|guidance_v_enter()">
-      <select cond="RCMode2()" exception="HOME"/>
-      <select cond="RCMode2() && DLModeNav()"/>
-      <control freq="32">
+    <mode name="NAV" start="guidance_h_nav_enter()|stabilization_attitude_enter()|guidance_v_z_enter()">
+      <select cond="RCMode2() && DLModeNav()" exception="HOME"/>
+      <control freq="NAV_FREQ">
         <call fun="nav_periodic_task()"/>
```

Comment:
I was just lazy, I can add it again, as long as guidance is not set from nav it's fine